server/src/user/user.service.ts

```diff
@@ -112,4 +112,43 @@ export class UserService {
       throw error;
     }
   }
+
+  async deleteUser(userId: number, houseId: number) {
+    try {
+      const userHouse = await this.prisma.userHouse.findFirst({
+        where: { userId, houseId },
+      });
+
+      if (!userHouse) {
+        throw new NotFoundException(
+          `User house not found for user ID ${userId} and house ID ${houseId}`,
+        );
+      }
+
+      // console.log('use House>>', userHouse);
+
+      if (userHouse.isAdmin) {
+        const otherAdminsCount = await this.prisma.userHouse.count({
+          where: { houseId, isAdmin: true, userId: { not: userId } },
+        });
+        if (otherAdminsCount === 0) {
+          throw new ForbiddenException(
+            'Cannot delete account. You are the only admin in this house.',
+          );
+        }
+      }
+
+      await this.prisma.userHouse.delete({
+        where: { id: userHouse.id },
+      });
+      await this.prisma.user.delete({
+        where: { id: userId },
+      });
```

Comment:
You don't have to delete from two tables. You can just delete from the user table.  Instead, please add `onDelete: Cascade` to the tables which are using user table reference! Like house delete!

Reply:
Fix it. 
fbc624aa4faf842e285187b7bf8baa88745c4d9c